ras_backstage/controllers/secure_messaging_controller.py

```diff
@@ -43,6 +44,30 @@ def get_message(encoded_jwt, message_id, is_draft):
     return json.loads(response.text)
 
 
+def get_thread_by_id(encoded_jwt, thread_id):
+    """
+    The method return all the message part of a thread.
+    :param encoded_jwt:  is the security token.
+    :param thread_id: is the id of the thread selected.
+    :return: a list of messages sent/received part of conversation thread
+    """
+
+    url = f"{app.config['RAS_SECURE_MESSAGING_SERVICE_THREAD']}{thread_id}"
+
+    headers = _create_authorization_header(encoded_jwt)
+    response = request_handler('GET', url, headers=headers)
+
+    if response.status_code != 200:
+        logger.error('Error retrieving the messages', thread_id=thread_id)
+        raise ApiError(url, response.status_code)
+
+    try:
+        return json.loads(response.text)
```

Comment:
return response.json()

Reply:
not convinced to do the right thing changing it. I do not see the perfomance and memery outome of this.

Reply:
changed